libs/checkpoint-validation/src/cli.ts

```diff
@@ -0,0 +1,109 @@
+
+export async function main() {
+  const moduleDirname = dirname(fileURLToPath(import.meta.url));
+
+  const builder = yargs();
+  await builder
+    .command(
+      "* <initializer-import-path> [filters..]",
+      "Validate a checkpoint saver",
+      {
+        builder: (args) => {
+          return args
+            .positional("initializerImportPath", {
+              type: "string",
+              describe:
+                "The import path of the CheckpointSaverTestInitializer for the checkpoint saver (passed to 'import()'). " +
+                "Must be the default export.",
+              demandOption: true,
+            })
+            .positional("filters", {
+              array: true,
+              choices: ["getTuple", "put", "putWrites", "list"],
+              default: [],
+              describe:
+                "Only run the specified suites. Valid values are 'getTuple', 'put', 'putWrites', and 'list'",
+              demandOption: false,
+            });
+        },
+        handler: async (
+          argv: ArgumentsCamelCase<{
+            initializerImportPath: string;
+            filters: string[];
+          }>
+        ) => {
+          const { initializerImportPath, filters } = argv;
+
+          let resolvedImportPath;
+
+          try {
+            resolvedImportPath = resolveImportPath(initializerImportPath);
+          } catch (e) {
+            console.error(
+              `Failed to resolve import path '${initializerImportPath}': ${e}`
+            );
+            process.exit(1);
+          }
+
+          let initializerExport: unknown;
+          try {
+            initializerExport = await dynamicImport(resolvedImportPath);
+          } catch (e) {
+            console.error(
+              `Failed to import initializer from import path '${initializerImportPath}' (resolved to '${resolvedImportPath}'): ${e}`
+            );
+            process.exit(1);
+          }
+
+          let initializer: CheckpointSaverTestInitializer<BaseCheckpointSaver>;
+          try {
+            initializer = checkpointSaverTestInitializerSchema.parse(
+              (initializerExport as { default?: unknown }).default ??
+                initializerExport
+            );
+            (
+              globalThis as GlobalThis
+            ).__langgraph_checkpoint_validation_initializer = initializer;
+            (
+              globalThis as GlobalThis
+            ).__langgraph_checkpoint_validation_filters =
+              filters as TestTypeFilter[];
+          } catch (e) {
+            console.error(
+              `Initializer imported from '${initializerImportPath}' does not conform to the expected schema. Make sure " +
+              "it is the default export, and that implements the CheckpointSaverTestInitializer interface. Error: ${e}`
+            );
+            process.exit(1);
+          }
+
+          await runCLI(
+            {
+              _: [],
+              $0: "",
```

Comment:
Could we avoid the need for the global by passing the initializer filename here as an extra option? Not sure what the API looks like

Reply:
I couldn't find any nice way to do this. If Jest had proper support for ESMs (and therefore allowed top-level `await` in test files), I could've made it much cleaner.

FWIW, `vitest` does support this, and [migration from Jest](https://vitest.dev/guide/migration#migrating-from-jest) isn't usually much of a drama. Definitely not advocating for that as a result of this PR, but it might be worth keeping it in mind if you hit more of these sort of issues in the future.